clientside/src/features/board/index.tsx

```diff
@@ -1,32 +1,147 @@
+import { useMemo } from "react";
 import styled from "styled-components";
 import Square from "./components/Square";
-import renderGridFromGridSize from "./functions/renderGridFromGridSize";
+import { Position, Side } from "./types/utility";
+import { UniBoardState } from "./types/board";
+import {
+    BattleshipAllyYard,
+    BattleshipDirection,
+    BattleshipPartRdState,
+    BattleshipPartType,
+    BattleshipStatus,
+    BattleshipYard,
+} from "./types/battleship";
 
-const GRID_SIZE = 10;
-const computedGrid = renderGridFromGridSize(GRID_SIZE);
+/**
+ * Overall Game State
+ * =====
+ * 0. MetaState:        Whose turn is it? How many turn have passed?
+ * 1. BoardState:       Which cells already get hit or missed?
+ * 2. BattleshipState:  Contain all battleships state, but enemy battleship
+ *                      are added here only when it's completely sunken
+ *                      meaning that its ready to be rendered. For example, at
+ *                      the begeinning of the game the enemy ships state is empty.
+ *                      Beginning state should also be randomized accordingly
+ *                      to the specified rules.
+ **/
 
-interface Props {
-    boardType: "player" | "enemy";
+type Props = AllyProps | EnemyProps;
+
+interface AllyProps {
+    board: UniBoardState;
+    boardType: Side.Ally;
+    shipYard: BattleshipAllyYard;
+}
+
+interface EnemyProps {
+    board: UniBoardState;
+    boardType: Side.Enemy;
+    shipYard: BattleshipYard;
 }
 
-const Board: React.FC<Props> = ({ boardType }) => {
-    const displayedSquares = computedGrid.map(({ position }) => {
+const Board: React.FC<Props> = ({ board, boardType, shipYard }) => {
+    const mappedBattleshipPart = useMemo(() => {
```

Comment:
Can we extract this into a function or many functions outside and use the dependencies as input; It'll be easier to look at